### PR TITLE
allow supported resource overrides in the limitranger plugin

### PR DESCRIPTION
**Pull request by pweil-**
@derekwaynecarr - found when implementing limit ranges for image resources post rebase.  PTAL

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
@pweil- can we combine `LimitFunc` and `SupportedResourcesFunc` into a common interface?  They basically need to be specified in lock-step so having them be provided separately feels error prone.

**Comment by derekwaynecarr** [comment]
@pweil- I am also fine if we defer that to a future PR, but put a TODO to merge the two into a common interface.

**Comment by pweil-** [comment]
Per discussion, updated to implement as an interface.  Also added the ability to ignore limits if required with a default implementation to not ignore any limits.

File: plugin/pkg/admission/limitranger/admission.go [hunk 1]
```diff
@@ -383,6 +375,38 @@ func sum(inputs []api.ResourceList) api.ResourceList {
...
+	case "pods":
+		return PodLimitFunc(limitRange, obj.(*api.Pod))
+	}
+	return nil
+}
+
+// SupportsAttributes ignores all calls that do not deal with pod resources since that is
+// all this supports now.  Also ignores any call that has a subresource defined.
+func (d *DefaultLimitRangerActions) SupportsAttributes(a admission.Attributes) bool {
+	if a.GetSubresource() != "" {
```
Comment (derekwaynecarr) [inline plugin/pkg/admission/limitranger/admission.go]: are we confident in our downstream use cases that we will not forget to include this line?

File: plugin/pkg/admission/limitranger/interfaces.go [hunk 2]
```diff
@@ -17,9 +17,18 @@ limitations under the License.
...
 
-// LimitFunc is a pluggable function to enforce limits on the object
-type LimitFunc func(limitRange *api.LimitRange, kind string, obj runtime.Object) error
+type LimitRangerActions interface {
+	// Limit is a pluggable function to enforce limits on the object.
+	Limit(limitRange *api.LimitRange, kind string, obj runtime.Object) error
+	// SupportsAttributes is a pluggable function to allow overridding what resources the limitranger
+	// supports.
+	SupportsAttributes(attr admission.Attributes) bool
+	// SupportsLimit is a pluggable function to allow ignoring limits that should not be applied
```
Comment (derekwaynecarr) [inline plugin/pkg/admission/limitranger/interfaces.go]: I assume this is when a limit is referencing a custom type?

File: plugin/pkg/admission/limitranger/admission.go (hunk 1 above)
Comment (pweil-) [inline plugin/pkg/admission/limitranger/admission.go]: I can certainly move it back into the main admit method if you think it's worth it.  

Here were the two downstream cases I found:
https://github.com/openshift/origin/pull/7989/files#diff-5fd1547857aea1ef0cdb18cb293f88faR210
https://github.com/openshift/origin/pull/7989/files#diff-c0324af3f9610f91b1fde15a63e35bf5R106

**Comment by derekwaynecarr** [comment]
LGTM

**Comment by derekwaynecarr** [comment]
actually, looks good, but need to see what travis is upset about...

File: plugin/pkg/admission/limitranger/interfaces.go (hunk 2 above)
Comment (pweil-) [inline plugin/pkg/admission/limitranger/interfaces.go]: Yes, used to filter limit ranges that only apply to images downstream: https://github.com/openshift/origin/pull/7989/files#diff-c0324af3f9610f91b1fde15a63e35bf5R118

**Comment by pweil-** [comment]
`No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.`

reopening to kick

**Pull request merged**